x/wasm/keeper/contract_keeper_test.go

```diff
@@ -169,3 +171,36 @@ func TestInstantiate2(t *testing.T) {
 		})
 	}
 }
+
+func TestQuerierError(t *testing.T) {
+	parentCtx, keepers := CreateTestInput(t, false, AvailableCapabilities)
+	parentCtx = parentCtx.WithGasMeter(storetypes.NewInfiniteGasMeter())
+
+	contract := InstantiateReflectExampleContract(t, parentCtx, keepers)
+
+	// this query will fail in the contract because there is no such reply
+	erroringQuery := testdata.ReflectQueryMsg{
+		SubMsgResult: &testdata.SubCall{
+			ID: 1,
+		},
+	}
+	// we make the reflect contract run the erroring query to check if our error stays
+	queryType := testdata.ReflectQueryMsg{
+		Chain: &testdata.ChainQuery{
+			Request: &wasmvmtypes.QueryRequest{
+				Wasm: &wasmvmtypes.WasmQuery{
+					Smart: &wasmvmtypes.SmartQuery{
+						ContractAddr: contract.Contract.String(),
+						Msg:          mustMarshal(t, erroringQuery),
+					},
+				},
+			},
+		},
+	}
+	query := mustMarshal(t, queryType)
+	_, err := keepers.WasmKeeper.QuerySmart(parentCtx, contract.Contract, query)
+	require.Error(t, err)
+
+	// we expect the contract's "reply 1 not found" to be in there
+	assert.Equal(t, "Generic error: Querier contract error: reply 1 not found: query wasm contract failed: query wasm contract failed", err.Error())
```

Comment:
What about having `assert.Contains(t, err.Error(), "reply 1 not found")` instead of `assert.Equal`?